File: java-kata/src/test/java/com/leetcode/ValidAnagramTest.java
package com.leetcode;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class ValidAnagramTest {

    private final ValidAnagram it = new ValidAnagram();

    @Test
    void test1() {
//        Input: s = "anagram", t = "nagaram"
//        Output: true

        var res = it.isAnagram("anagram", "nagaram");
        assertThat(res)
                .isTrue();
    }

    @Test
    void test2() {
//        Input: s = "rat", t = "car"
//        Output: false

        var res = it.isAnagram("rat", "car");
        assertThat(res)
                .isFalse();
    }

}

File: kotlin-kata/src/test/kotlin/com/leetcode/ValidAnagramTest.kt
package com.leetcode

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class ValidAnagramTest {
    private val it = ValidAnagram()

    @Test
    fun test1() {
//        Input: s = "anagram", t = "nagaram"
//        Output: true

        val res = it.isAnagram("anagram", "nagaram")
        Assertions.assertThat(res)
            .isTrue()
    }

    @Test
    fun test2() {
//        Input: s = "rat", t = "car"
//        Output: false

        val res = it.isAnagram("rat", "car")
        Assertions.assertThat(res)
            .isFalse()
    }
}